File: JavaLox/src/com/craftinginterpreters/lox/LoxLinkedList.java
package com.craftinginterpreters.lox;

import java.util.LinkedList;
import java.util.List;

/**
 *
 * @author brusc
 */
class LoxLinkedList extends LoxInstance {
   
   final List<Object> elements;
   
   LoxLinkedList() {
      super(null);
      elements = new LinkedList<>();      
   }
   
   @Override
   Object get(Token name) {
      if (name.lexeme.equals("get")) {
         return new LoxLinkedListGet(this);
         
      } else if (name.lexeme.equals("set")) {
         return new LoxLinkedListSet(this);
         
      } else if (name.lexeme.equals("add")) {
         return new LoxLinkedListAdd(this);
         
      } else if (name.lexeme.equals("length")) {
         return (double) elements.size();
         
      }

      throw new RuntimeError(name, "Undefined property '" + name.lexeme + "'.");
   }

   @Override
   void set(Token name, Object value) {
      throw new RuntimeError(name, "Can't add properties to arrays.");
   }

   @Override
   public String toString() {
      StringBuilder buffer = new StringBuilder();
      buffer.append("(");
      for (int i = 0; i < elements.size(); i++) {
         if (i != 0) {
            buffer.append(", ");
         }
         buffer.append(elements.get(i));
      }
      buffer.append(")");
      return buffer.toString();
   }   
}

File: CsLox/com/craftinginterpreters/lox/LoxLinkedList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.craftinginterpreters.lox
{
    internal class LoxLinkedList : LoxInstance
    {
        internal readonly List<Object> elements;

        internal LoxLinkedList() : base(null)
        {
            elements = new List<Object>();
        }

        internal override Object get(Token name)
        {
            if (name.lexeme.Equals("get"))
            {
                return new LoxLinkedListGet(this);
            }
            else if (name.lexeme.Equals("set"))
            {
                return new LoxLinkedListSet(this);
            }
            else if (name.lexeme.Equals("add"))
            {
                return new LoxLinkedListAdd(this);
            }
            else if (name.lexeme.Equals("length"))
            {
                return (double)elements.Count;
            }

            throw new RuntimeError(name, "Undefined property '" + name.lexeme + "'.");
        }

        internal override void set(Token name, Object value)
        {
            throw new RuntimeError(name, "Can't add properties to arrays.");
        }

        public override string ToString()
        {
            StringBuilder buffer = new StringBuilder();
            buffer.Append("(");
            for (int i = 0; i < elements.Count; i++)
            {
                if (i != 0)
                {
                    buffer.Append(", ");
                }
                buffer.Append(elements[i]);
            }
            buffer.Append(")");
            return buffer.ToString();
        }
    }
}
